Assets/Scripts/Scenes/InGame/Enemy/Trace/Feature/FeatureView.cs

```diff
@@ -82,19 +91,81 @@ public void Grow()
 
         public void TryInteract()
         {
+            //5マス,1マスいないのインタラクトできるオブジェクトをそれぞれ格納
             nearStageObject = _stageInteracts.Where(target => Vector3.Distance(target.transform.position, _enemy.transform.position) < RANGE).ToArray();
+            moreNearStageObject = _stageInteracts.Where(target => Vector3.Distance(target.transform.position, _enemy.transform.position) < NEAR_RANGE).ToArray();
             if (nearStageObject.Length > 0)
             {
                 interactTarget = nearStageObject[UnityEngine.Random.Range(0, nearStageObject.Length)];
+                nearInteractTarget = nearStageObject[UnityEngine.Random.Range(0, moreNearStageObject.Length)];
                 if (interactTarget.TryGetComponent(out IInteractable act))
                 {
-                    act.Intract(null, true);
+                    randomValue = UnityEngine.Random.Range(0, 100);
+                    //randomValue < 10 で10%の確率でイントラクトされるよう設定
+                    if (interactTarget.TryGetComponent(out StageLight light) && randomValue < 10)
+                    {
+                        Debug.Log("照明にインタラクトしました");
+                        act.Intract(null, true);
+                    }
+                    else if (interactTarget.TryGetComponent(out StageDoor door_open) && door_open.ReturnIsOpen == false)
+                        act.Intract(null, true);
+                    //randomValue < 5 で5%の確率でイントラクトされるよう設定
+                    if ((nearInteractTarget.TryGetComponent(out StageDoor door_close) || nearInteractTarget.TryGetComponent(out StageRack Rack)) && randomValue < 5)
+                    {
+                        Debug.Log("ドアか棚にインタラクトしました");
+                        act.Intract(null, true);
+                    }
                 }
             }
         }
         public void InstanceTrackSprite()
         {
-            GameObject trackSprite = Instantiate(_trackSprite, _enemy.transform.position, Quaternion.identity);
+            Debug.Log("動いている");
+            //近くの壁を格納
+            LayerMask wallMask = LayerMask.GetMask("Wall");
+            RaycastHit hit;
+            traceSelect = UnityEngine.Random.Range(0, _maxTraceType);
+            //プレイヤーの左右3以内にある壁を取得
+            Vector3 nearWallCheck = new Vector3(_enemy.transform.position.x - 3, _enemy.transform.position.y + 2, _enemy.transform.position.z);
+            Ray ray = new Ray(nearWallCheck, direction_wall);
+            // レイをデバッグ表示
+            Debug.DrawRay(ray.origin, ray.direction * 6.0f, Color.red, 2.0f);
```

Comment:
```suggestion
#if UNITY_EDITOR
            Debug.DrawRay(ray.origin, ray.direction * 6.0f, Color.red, 2.0f);
#endif
```
エディタのみで動作するようにしたいので、上記のように修正してください。